From 05d02d8db0b3a7fd7634f7a0bd35ad96da469fb4 Mon Sep 17 00:00:00 2001
From: Lennart Poettering <lennart@poettering.net>
Date: Thu, 09 Feb 2012 22:25:57 +0000
Subject: main: fix reexecution of systemd, use right path to binary

---
diff --git a/Makefile.am b/Makefile.am
index c5b819e..122acfa 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -86,7 +86,7 @@ AM_CPPFLAGS = \
 	-DUSER_CONFIG_UNIT_PATH=\"$(pkgsysconfdir)/user\" \
 	-DUSER_DATA_UNIT_PATH=\"$(userunitdir)\" \
 	-DSYSTEMD_CGROUP_AGENT_PATH=\"$(rootlibexecdir)/systemd-cgroups-agent\" \
-	-DSYSTEMD_BINARY_PATH=\"$(rootbindir)/systemd\" \
+	-DSYSTEMD_BINARY_PATH=\"$(rootlibexecdir)/systemd\" \
 	-DSYSTEMD_SHUTDOWN_BINARY_PATH=\"$(rootlibexecdir)/systemd-shutdown\" \
 	-DSYSTEMCTL_BINARY_PATH=\"$(rootbindir)/systemctl\" \
 	-DSYSTEMD_TTY_ASK_PASSWORD_AGENT_BINARY_PATH=\"$(rootbindir)/systemd-tty-ask-password-agent\" \
diff --git a/TODO b/TODO
index ed857f8..5ee199f 100644
--- a/TODO
+++ b/TODO
@@ -21,6 +21,8 @@ Bugfixes:
 
 Features:
 
+* support closing all fds via RLIMIT_NOFILE instead of /proc, in order to make chroot stuff work.
+
 * add interface to allow immediate rotation of the journal, and even flushing.
 
 * don't log coredumps of PID 1 into the journal
--
cgit v0.9.0.2-2-gbebe
