From 23218f74c103cdd8205b68c1b1ab8416006f13af Mon Sep 17 00:00:00 2001
From: Niels Breet <niels.breet@jolla.com>
Date: Mon, 6 Jul 2020 12:25:23 +0300
Subject: [PATCH] [aarch64] Force udev path. Contributes to JB#49681

---
 Makefile.am | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/Makefile.am b/Makefile.am
index c2973c0..2a969fe 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -100,7 +100,8 @@ systemshutdowndir=$(rootlibexecdir)/system-shutdown
 systemsleepdir=$(rootlibexecdir)/system-sleep
 systemunitdir=$(rootprefix)/lib/systemd/system
 systempresetdir=$(rootprefix)/lib/systemd/system-preset
-udevlibexecdir=$(rootprefix)/lib/udev
+#Force this to udevlibexecdir to /lib/udev for now, to make transition easier
+udevlibexecdir=/lib/udev
 udevhomedir=$(udevlibexecdir)
 udevrulesdir=$(udevlibexecdir)/rules.d
 udevhwdbdir=$(udevlibexecdir)/hwdb.d
-- 
1.9.1

