diff --git a/src/core/mount.c b/src/core/mount.c
index b4721b0c2e..49d7bebb41 100644
--- a/src/core/mount.c
+++ b/src/core/mount.c
@@ -486,7 +486,7 @@ static int mount_add_default_ordering_dependencies(
                 before = SPECIAL_LOCAL_FS_TARGET;
         }
 
-        if (!mount_is_nofail(m)) {
+        if (!mount_is_nofail(m) && !path_equal(m->where, "/home")) {
                 r = unit_add_dependency_by_name(UNIT(m), UNIT_BEFORE, before, true, mask);
                 if (r < 0)
                         return r;
diff --git a/src/fstab-generator/fstab-generator.c b/src/fstab-generator/fstab-generator.c
index 9b32383a76..fa6260de0b 100644
--- a/src/fstab-generator/fstab-generator.c
+++ b/src/fstab-generator/fstab-generator.c
@@ -437,7 +437,7 @@ static int add_mount(
 
         /* Order the mount unit we generate relative to the post unit, so that DefaultDependencies= on the
          * target unit won't affect us. */
-        if (post && !FLAGS_SET(flags, MOUNT_NOFAIL))
+        if (post && !FLAGS_SET(flags, MOUNT_NOFAIL) && !path_equal(where, "/home"))
                 fprintf(f, "Before=%s\n", post);
 
         if (passno != 0) {

