From e24b4c602f3e70ee36bf749fd0d51b2740992e0f Mon Sep 17 00:00:00 2001
From: Pekka Lundstrom <pekka.lundstrom@jollamobile.com>
Date: Tue, 29 Apr 2014 09:34:07 +0300
Subject: [PATCH] Count restarts when doing restart, not in start

Signed-off-by: Pekka Lundstrom <pekka.lundstrom@jollamobile.com>
---
 src/core/service.c |   17 +++++++++--------
 1 file changed, 9 insertions(+), 8 deletions(-)

diff --git a/src/core/service.c b/src/core/service.c
index 35053be..9d5f4c6 100644
--- a/src/core/service.c
+++ b/src/core/service.c
@@ -1728,6 +1728,8 @@ fail:
         service_enter_dead(s, SERVICE_FAILURE_RESOURCES, true);
 }
 
+static int service_start_limit_test(Service *s);
+
 static void service_enter_restart(Service *s) {
         _cleanup_bus_error_free_ sd_bus_error error = SD_BUS_ERROR_NULL;
         int r;
@@ -1745,6 +1747,13 @@ static void service_enter_restart(Servic
                 return;
         }
 
+        /* Make sure we don't start services too frequently */
+        r = service_start_limit_test(s);
+        if (r < 0) {
+                service_enter_dead(s, SERVICE_FAILURE_START_LIMIT, false);
+                return;
+        }
+
         /* Any units that are bound to this service must also be
          * restarted. We use JOB_RESTART (instead of the more obvious
          * JOB_START) here so that those dependency jobs will be added
@@ -1898,7 +1907,6 @@ static int service_start_limit_test(Serv
 
 static int service_start(Unit *u) {
         Service *s = SERVICE(u);
-        int r;
 
         assert(s);
 
@@ -1925,13 +1933,6 @@ static int service_start(Unit *u) {
 
         assert(IN_SET(s->state, SERVICE_DEAD, SERVICE_FAILED));
 
-        /* Make sure we don't enter a busy loop of some kind. */
-        r = service_start_limit_test(s);
-        if (r < 0) {
-                service_enter_dead(s, SERVICE_FAILURE_START_LIMIT, false);
-                return r;
-        }
-
         s->result = SERVICE_SUCCESS;
         s->reload_result = SERVICE_SUCCESS;
         s->main_pid_known = false;

-- 
1.7.9.5

