From b0903bb585d2158f78a4cf603fde4aed5474e2b9 Mon Sep 17 00:00:00 2001
From: Yu Watanabe <watanabe.yu+github@gmail.com>
Date: Wed, 2 May 2018 17:05:51 +0900
Subject: [PATCH] meson: drop 'name' argument in cc.has_argument() (#8878)

---
 meson.build | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/meson.build b/meson.build
index 93cbabf260..0086f77483 100644
--- a/meson.build
+++ b/meson.build
@@ -430,8 +430,7 @@ endforeach
 if get_option('buildtype') != 'debug'
         foreach arg : ['-ffunction-sections',
                        '-fdata-sections']
-                if cc.has_argument(arg,
-                                   name : '@0@ is supported'.format(arg))
+                if cc.has_argument(arg)
                         add_project_arguments(arg, language : 'c')
                 endif
         endforeach
-- 
2.26.2

