diff --git a/Makefile.am b/Makefile.am
index 8d9c587..5622ffa 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -109,6 +109,7 @@ kernelinstalldir = $(prefix)/lib/kernel/
 factory_etcdir = $(datadir)/factory/etc
 factory_pamdir = $(datadir)/factory/etc/pam.d
 bootlibdir = $(prefix)/lib/systemd/boot/efi
+testbindir = /opt/tests/systemd-tests/bin
 
 # And these are the special ones for /
 rootprefix=@rootprefix@
@@ -147,11 +148,12 @@ tests=
 manual_tests =
 TEST_EXTENSIONS = .py
 PY_LOG_COMPILER = $(PYTHON)
+noinst_PROGRAMS =
 if ENABLE_TESTS
-noinst_PROGRAMS = $(manual_tests) $(tests)
+testbin_PROGRAMS = $(manual_tests) $(tests)
 TESTS = $(tests)
 else
-noinst_PROGRAMS =
+testbin_PROGRAMS =
 TESTS =
 endif
 udevlibexec_PROGRAMS =
