Remove pam_selinux.so since we don't have it.

diff --git a/src/login/systemd-user.in b/src/login/systemd-user.in
--- a/src/login/systemd-user.in
+++ b/src/login/systemd-user.in
@@ -10,8 +10,8 @@ account sufficient pam_unix.so no_pass_expiry
 account required pam_permit.so
 
 {% if HAVE_SELINUX %}
-session required pam_selinux.so close
-session required pam_selinux.so nottys open
+#session required pam_selinux.so close
+#session required pam_selinux.so nottys open
 {% endif %}
 session required pam_loginuid.so
 session optional pam_keyinit.so force revoke
