From 8f164fb28a8b8c415143245a6a543c27a0deea1f Mon Sep 17 00:00:00 2001
From: Pekka Lundstrom <pekka.lundstrom@jollamobile.com>
Date: Mon, 17 Jun 2013 16:06:02 +0300
Subject: [PATCH] Remove display-manager.service

Signed-off-by: Pekka Lundstrom <pekka.lundstrom@jollamobile.com>
---
 units/graphical.target | 1 -
 1 file changed, 1 deletion(-)

diff --git a/units/graphical.target b/units/graphical.target
index 4b2087fa1b..a9031e502a 100644
--- a/units/graphical.target
+++ b/units/graphical.target
@@ -11,7 +11,6 @@
 Description=Graphical Interface
 Documentation=man:systemd.special(7)
 Requires=multi-user.target
-Wants=display-manager.service
 Conflicts=rescue.service rescue.target
 After=multi-user.target rescue.service rescue.target display-manager.service
 AllowIsolate=yes
