diff --git a/Makefile.am b/Makefile.am
index 8d9c587..7a91eaa 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -70,7 +70,7 @@ dbussessionservicedir=@dbussessionservic
 dbussystemservicedir=@dbussystemservicedir@
 pamlibdir=@pamlibdir@
 pamconfdir=@pamconfdir@
-pkgconfigdatadir=$(datadir)/pkgconfig
+pkgconfigdatadir=$(libdir)/pkgconfig
 pkgconfiglibdir=$(libdir)/pkgconfig
 polkitpolicydir=$(datadir)/polkit-1/actions
 bashcompletiondir=@bashcompletiondir@
