diff --git a/Makefile.am b/Makefile.am
index 8d9c587..7a91eaa 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -65,7 +65,7 @@ dbussystemservicedir=@dbussystemservicedir@
 dbusinterfacedir=@dbusinterfacedir@
 pamlibdir=@pamlibdir@
 pamconfdir=@pamconfdir@
-pkgconfigdatadir=$(datadir)/pkgconfig
+pkgconfigdatadir=$(libdir)/pkgconfig
 pkgconfiglibdir=$(libdir)/pkgconfig
 polkitpolicydir=$(datadir)/polkit-1/actions
 bashcompletiondir=@bashcompletiondir@
@@ -2228,7 +2228,7 @@ udevconfdir = $(sysconfdir)/udev
 dist_udevconf_DATA = \
 	src/udev/udev.conf
 
-sharepkgconfigdir = $(datadir)/pkgconfig
+sharepkgconfigdir = $(libdir)/pkgconfig
 sharepkgconfig_DATA = \
 	src/udev/udev.pc
 
